include/libopencm3/stm32/h7/rcc.h

```diff
@@ -200,6 +439,37 @@ enum rcc_osc {
   RCC_LSI
 };
 
+enum rcc_sysclk_mux {
+  RCC_SYSCLK_PLL,
+  RCC_SYSCLK_HSE,
+  RCC_SYSCLK_HSI,
+};
+
+enum rcc_pll_mux {
+  RCC_PLL_HSI = RCC_PLLCKSELR_PLLSRC_HSI,
+  RCC_PLL_HSE = RCC_PLLCKSELR_PLLSRC_HSE
+};
+
+/** PLL Configuration structure. */
+struct rcc_pll_config {
+  uint32_t hse_frequency;           /**< User configured external crystal frequency. */
+  enum rcc_sysclk_mux sysclk_mux;   /**< SYSCLK source input selection. */
+  enum rcc_pll_mux pll_mux;         /**< PLL source input selection. */
+  struct pll_config {
+    uint8_t divm;                   /**< Pre-divider value for each PLL. 0-64 integers. */
+    uint16_t divn;                  /**< Multiplier, 0-512 integer. */
+    uint8_t divp;                   /**< Post divider for PLLP clock. */
+    uint8_t divq;                   /**< Post divider for PLLQ clock. */
+    uint8_t divr;                   /**< Post divider for PLLR clock. */
+  } pll1, pll2, pll3;               /**< PLL1-PLL3 configurations. */
+  uint32_t d1cfg_core_prescale;     /**< Core prescaler for domain 1. */
+  uint32_t d1cfg_hclk3_prescale;    /**< HCLK3 prescaler for domain 1. */
+  uint32_t d1cfg_pclk3_prescale;    /**< APB3 Peripheral prescaler for domain 1. */
+  uint32_t d2cfg_pclk1_prescale;    /**< APB1 Peripheral prescaler for domain 2. */
+  uint32_t d2cfg_pclk2_prescale;    /**< APB2 Peripheral prescaler for domain 2. */
+  uint32_t d3cfg_pclk4_prescale;    /**< APB4 Peripheral prescaler for domain 3. */
+};
```

Comment:
This is _similar_ but not to the existing rcc_clock_scale structs.  I know it does more, but it brings in all of the pclk prescaler selection, makes it massively incompatible.  G0 and L4 are probably closes here too, where it's set by subsequent calls?